VideoApi/VideoApi/Health/WowzaHealthCheck.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using VideoApi.Services.Contracts;
+
+namespace VideoApi.Health;
+
+public class WowzaHealthCheck : IHealthCheck
+{
+    private readonly IAudioPlatformService _audioPlatformService;
+
+    public WowzaHealthCheck(IAudioPlatformService audioPlatformService)
+    {
+        _audioPlatformService = audioPlatformService;
+    }
+
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = new CancellationToken())
+    {
+        try
+        {
+            var healthyResult = await _audioPlatformService.GetDiagnosticsAsync();
+            return healthyResult ? HealthCheckResult.Healthy() : HealthCheckResult.Degraded("Wowza is not healthy");
+        }
+        catch (Exception exception)
+        {
+            return HealthCheckResult.Degraded(exception.Message, exception);
```

Comment:
Exceptions are already being handled in the GetDiagnosticsAsync method, which would return false in that instance. This catch/all other exceptionw - would only ever be hit, if their was some sort of error that wasn't related to the request to wowza, but within the code. Meaning it might not be wowza which is degraded. Not really a big deal and makes sense to have a safety net, but thought I'd highlight it anyway.

Reply:
I thought in the meeting we agreed health checks shouldn't tank the pod. If there's an unhandled exception in the app code, that would make the liveness fail but if there's an issue with diagnostics then restarting the pod won't fix it but degraded will get the attention from Ops team

Reply:
Yeah that's fine, my point was more leaning towards the health check showing wowza as being degarded when actually it might not be, but our code, and therfore getting us to investigate the wrong thing. But actually we're returning the message and the exception in the response, so we'd be able to work that out; so yeah my point doesnt make too much difference either way... So just take your approval and get out of my office, damn it. 👈🏻